lua/nvim-treesitter/ts_utils.lua

```diff
@@ -187,12 +182,68 @@ function M.is_in_node_range(node, line, col)
   end
 end
 
+function M.get_node_range(node_or_range)
+  if type(node_or_range) == 'table' then
+    return unpack(node_or_range)
+  else
+    return node_or_range:range()
+  end
+end
+
 function M.node_to_lsp_range(node)
-  local start_line, start_col, end_line, end_col = node:range()
+  local start_line, start_col, end_line, end_col = M.get_node_range(node)
   local rtn = {}
   rtn.start = { line = start_line, character = start_col }
   rtn['end'] = { line = end_line, character = end_col }
   return rtn
 end
 
+function M.swap_nodes(node_or_range1, node_or_range2, bufnr, cursor_to_second)
+  if not node_or_range1 or not node_or_range2 then return end
+  local range1 = M.node_to_lsp_range(node_or_range1)
+  local range2 = M.node_to_lsp_range(node_or_range2)
+
+  local text1 = M.get_node_text(node_or_range1)
+  local text2 = M.get_node_text(node_or_range2)
+
+  local edit1 = { range = range1, newText = table.concat(text2, '\n') }
+  local edit2 = { range = range2, newText = table.concat(text1, '\n') }
+  vim.lsp.util.apply_text_edits({edit1, edit2}, bufnr)
+
+  if cursor_to_second then
+    -- Set the item in jump list
```

Comment:
this is not needed either imo

Reply:
before we had discussion with @vigoux to create a helper function for this.